### PR TITLE
Add generate, generate_n and trasform algorithms as tested API

### PR DESCRIPTION
**Pull request by rarutyun**
This is the continuation of #356 review.

Moved from fork to branch and rebase.

### CODE REVIEW

**Review by SergeyKopienko — APPROVED** [review]
LGTM

**Comment by akukanov** [comment]
I am fine with the comments in the code (which Ruslan asked me to review). The rest of the patch I have not reviewed.

File: documentation/library_guide/api_for_sycl_kernels/tested_standard_cpp_api.rst [hunk 1]
```diff
@@ -444,6 +444,12 @@ C++ Standard API                      libstdc++  libc++     MSVC
 ``std::push_heap``                    Tested     Tested     Tested
 ------------------------------------- ---------- ---------- ----------
 ``std::pop_heap``                     Tested     Tested     Tested
+------------------------------------ ---------- ---------- ----------
+``std::generate``                     Tested     Tested     Tested
```
Comment (SergeyKopienko) [inline documentation/library_guide/api_for_sycl_kernels/tested_standard_cpp_api.rst:448]: Please move std::generate row to upper for alphabetical order as before

Comment (rarutyun) [reply documentation/library_guide/api_for_sycl_kernels/tested_standard_cpp_api.rst:448]: There is no alphabetical order in this file.

**Review by SergeyKopienko — APPROVED** [review]
LGTM

**Review by akukanov — APPROVED** [review]

**Pull request merged**